python/base/Dockerfile.python

```diff
@@ -0,0 +1,51 @@
+ARG CIRCLE_PULL_REQUEST
+FROM debian:bullseye-slim AS builder
+
+# Install dependencies for building Python
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    wget \
+    cmake \
+    build-essential \
+    libssl-dev \
+    zlib1g-dev \
+    libncurses5-dev \
+    libncursesw5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libgdbm-dev \
+    libdb5.3-dev \
+    libbz2-dev \
+    libexpat1-dev \
+    liblzma-dev \
+    tk-dev \
+    libffi-dev \
+    uuid-dev \
+    ca-certificates \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Download and extract the Python source code
+WORKDIR /tmp
+ARG PYTHON_VERSION_PATCH
+RUN wget -qO- "https://www.python.org/ftp/python/${PYTHON_VERSION_PATCH}/Python-${PYTHON_VERSION_PATCH}.tgz" | tar xvz \
+    && cd "/tmp/Python-${PYTHON_VERSION_PATCH}" \
+    && ./configure --enable-optimizations --with-ensurepip=install \
+    && make -j "$(nproc)" \
+    && make altinstall
+
+
+FROM deepnote/python:base${CIRCLE_PULL_REQUEST:+-ra-${CIRCLE_PULL_REQUEST##*/}}
+# Copy Python from the builder stage
+ARG PYTHON_VERSION
+# Layers will be different between python versions from here onwards because of the build-arg
+
+COPY --from=builder "/usr/local/bin/python${PYTHON_VERSION}" "/usr/local/bin/python${PYTHON_VERSION}"
+COPY --from=builder "/usr/local/bin/pip${PYTHON_VERSION}" "/usr/local/bin/pip${PYTHON_VERSION}"
+COPY --from=builder "/usr/local/lib/python${PYTHON_VERSION}" "/usr/local/lib/python${PYTHON_VERSION}"
+
+RUN update-alternatives --install /usr/bin/python python "/usr/local/bin/python${PYTHON_VERSION}" 1
+RUN update-alternatives --install /usr/bin/pip pip "/usr/local/bin/pip${PYTHON_VERSION}" 1
+
```

Comment:
This logic is somehow copied from official Python because I'm not sure if it actually recommends a version if installation does not have some side effects.

Reply:
But if you are sure that it is fine, I'm okay with that. It's working, so I think we are good.

Reply:
Ok, let's discuss this. I think I don't understand how this can work.